app/components/MainComponent.vue

```diff
@@ -104,17 +112,103 @@ export default defineNuxtComponent({
       clear();
+    },
+    async handleDateRangeChange(dateRange: { since?: string; until?: string; description: string }) {
+      this.dateRangeDescription = dateRange.description;
+      this.isLoading = true;
+      
+      try {
+        // Build query parameters
+        const params = new URLSearchParams();
+        if (dateRange.since) params.append('since', dateRange.since);
+        if (dateRange.until) params.append('until', dateRange.until);
+        
+        const queryString = params.toString();
+        const apiUrl = queryString ? `/api/metrics?${queryString}` : '/api/metrics';
+        
+        const response = await $fetch(apiUrl) as MetricsApiResponse;
+        
+        this.metrics = response.metrics || [];
+        this.originalMetrics = response.usage || [];
+        this.metricsReady = true;
+        this.apiError = undefined;
+      } catch (error: any) {
+        this.processError(error);
+      } finally {
+        this.isLoading = false;
+      }
+    },
+    processError(error: H3Error) {
+      console.error(error || 'No data returned from API');
+      // Check the status code of the error response
+      if (error.statusCode) {
+        switch (error.statusCode) {
+          case 401:
+            this.apiError = '401 Unauthorized access returned by GitHub API - check if your token in the .env (for local runs). Check PAT token and GitHub permissions.';
+            break;
+          case 404:
+            this.apiError = `404 Not Found - is the ${this.config?.public?.scope || ''} org:"${this.config?.public?.githubOrg || ''}" ent:"${this.config?.public?.githubEnt || ''}" team:"${this.config?.public?.githubTeam}" correct? ${error.message}`;
+            break;
+          case 422:
+            this.apiError = `422 Unprocessable Entity - Is the Copilot Metrics API enabled for the Org/Ent? ${error.message}`;
+            break;
+          case 500:
+            this.apiError = `500 Internal Server Error - most likely a bug in the app. Error: ${error.message}`;
+            break;
+          default:
+            this.apiError = `${error.statusCode} Error: ${error.message}`;
+            break;
+        }
+      }
     }
   },
 
   data() {
     return {
       tabItems: ['languages', 'editors', 'copilot chat', 'seat analysis', 'api response'],
-      tab: null
+      tab: null,
+      dateRangeDescription: 'Over the last 28 days',
+      isLoading: false,
+      metricsReady: false,
+      metrics: [] as Metrics[],
+      originalMetrics: [] as CopilotMetrics[],
+      seatsReady: false,
+      seats: [] as Seat[],
+      apiError: undefined as string | undefined,
+      config: null as any
     }
   },
   created() {
     this.tabItems.unshift(this.itemName);
+    this.config = useRuntimeConfig();
+  },
+  async mounted() {
+    // Load initial data
+    try {
+      const { data: metricsData, error: metricsError } = await this.metricsFetch;
+      if (metricsError || !metricsData) {
```

Comment:
this looks like a bug - it should be 
`metricsError.value` and `metricsData.value`